app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_post, only: %i[edit update destroy show]
+  before_action :correct_user, only: %i[edit update destroy]
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to root_path
+      flash[:success] = 'Post created successfully'
+    else
+      flash.now[:danger] = 'Post creation failed. Please enter valid information'
+      render 'new'
+    end
+  end
+
+  def edit; end
+
+  def show; end
+
+  def update
+    if @post.update(post_params)
+      flash[:success] = 'Post saved successfully'
+      redirect_to root_path
+    else
+      flash.now[:danger] = 'Post update failed. Please enter valid information'
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @post.destroy
+    flash[:success] = 'Post deleted successfully'
+    redirect_to root_path
+  end
+
+  def index
+    @posts = Post.all
```

Comment:
- [ ] PostsController#index displays posts in a timeline -- We can achieve this by applying a [default scope](https://apidock.com/rails/ActiveRecord/Base/default_scope/class) of retrieving posts with specific order to our `post` model

Reply:
We've implemented the requested change.